### PR TITLE
Use simplified version of GitHub button JS.

### PR DESCRIPTION
**Pull request by simi**
- :thinking: :eyes: :white_check_mark: https://github.com/rubygems/rubygems.org/security/code-scanning/34 and similar reports

### CODE REVIEW

**Comment by simi** [comment]
:information_source: seems this approach fixes CodeQL problems with the `github_buttons.js` file.

https://github.com/rubygems/rubygems.org/pull/3348/checks?check_run_id=10685669277

I have stripped it down to the minimal version needed for our simple purpose and changed from passing attributes as query string of one data attribute to individual data attributes.

## before (see `data-params`)

```html
<span class="github-btn github-stargazers github-btn-large" data-params="user=axgle&amp;repo=a&amp;type=star&amp;count=true&amp;size=large" %="">
  <a class="gh-btn" href="https://github.com/axgle/a" rel="noopener noreferrer" target="_blank" aria-label="Star axgle/a on GitHub">
    <span class="gh-ico" aria-hidden="true"></span>
    <span class="gh-text">Star</span>
  </a>
  <a class="gh-count" href="https://github.com/axgle/a/stargazers" rel="noopener noreferrer" target="_blank" aria-label="7 stargazers on GitHub" style="display: block;">7</a>
</span>
```

## after (see `data-user`, `data-repo`, ...)

```html
<span class="github-btn github-stargazers github-btn-large" data-user="axgle" data-repo="a" data-type="star" data-count="true" data-size="large">
  <a class="gh-btn" href="https://github.com/axgle/a" rel="noopener noreferrer" target="_blank" aria-label="Star axgle/a on GitHub">
    <span class="gh-ico" aria-hidden="true"></span>
    <span class="gh-text">Star</span>
  </a>
  <a class="gh-count" href="https://github.com/axgle/a/stargazers" rel="noopener noreferrer" target="_blank" aria-label="7 stargazers on GitHub" style="display: block;">7</a>
</span>
```

**Comment by simi** [comment]
Btw. I'm not sure any real problem actually got fixed, but this seems a simpler approach anyway. We will need to find out better way to maintain our JavaScript stack sooner or later anyway. This is just intermediate solution.

File: app/helpers/rubygems_helper.rb [hunk 1]
```diff
@@ -139,6 +139,14 @@ def link_to_github(rubygem)
 
   def github_params(rubygem)
     link = link_to_github(rubygem)
-    "user=#{link.path.split('/').second}&repo=#{link.path.split('/').third}&type=star&count=true&size=large" if link
+    return unless link
```
Comment (colby-swandale) [inline app/helpers/rubygems_helper.rb:142]: Should we also take into consideration an empty string as well?

```suggestion
    return unless link.present?
```

File: vendor/assets/javascripts/github_buttons.js [hunk 2]
```diff
@@ -12,177 +12,89 @@
...
-      return;
-    }
-
-    switch (type) {
-      case 'watch':
-        if (v === '2') {
-          counter.textContent = obj.data.subscribers_count && addCommas(obj.data.subscribers_count);
-          counter.setAttribute('aria-label', counter.textContent + ' watchers' + LABEL_SUFFIX);
+    function jsonp(path) {
+      var xhr = new XMLHttpRequest();
```
Comment (colby-swandale) [inline vendor/assets/javascripts/github_buttons.js:31]: Not saying this needs to be solved right now in the PR, but I would be very keen to get rubygems.org away from `XMLHttpRequest`. [fetch](https://caniuse.com/fetch) is now covered by the vast majority of browsers and there's plenty of libraries that put it behind a nice abstraction like https://github.com/sindresorhus/ky

File: vendor/assets/javascripts/github_buttons.js [hunk 3]
```diff
@@ -12,177 +12,89 @@
...
+
+    // Parameters
+    var user = mainButton.dataset.user;
+    var repo = mainButton.dataset.repo;
+    var type = mainButton.dataset.type;
+    var count = mainButton.dataset.count;
+    var size = mainButton.dataset.size;
+
+    // Constants
+    var LABEL_SUFFIX = ' on GitHub';
```
Comment (colby-swandale) [inline vendor/assets/javascripts/github_buttons.js:60]: `var` allows for mutation, could we use `const` instead here?

**Review by colby-swandale — APPROVED** [review]
I haven't run this on my machine to test, but this looks good to me. Just some small notes/questions.

File: app/helpers/rubygems_helper.rb [hunk 4]
```diff
@@ -139,6 +139,14 @@ def link_to_github(rubygem)
 
   def github_params(rubygem)
```
Comment (jenshenny) [inline app/helpers/rubygems_helper.rb:140]: Should we add a unit test for this?

File: vendor/assets/javascripts/github_buttons.js (hunk 3 above)
Comment (jenshenny) [reply vendor/assets/javascripts/github_buttons.js:60]: Rubygems.org solely uses `var` at the moment 😭  so out of convention, I'm ok to keep it as is.
I think we should go through the js files and convert vars into `let`s and `const`s some time.

**Review by jenshenny — APPROVED** [review]
LGTM! 🚀  Tested it out locally

**Comment by simi** [comment]
If I remember well, @colby-swandale was about to prepare modern JS stack prototype. Let's keep the change minimal for now leaving it as is.

**Pull request merged**